language/resource-api/README.md

```diff
@@ -393,13 +403,15 @@ class Puppet::Provider::AptKey::AptKey
   end
 ```
 
-> Note: It is recommended to create the command in the `initialize` function of the provider, and store them in a member named after the command, with the `_cmd` suffix. This makes it easy to re-use common settings throughout the provider.
+> Note: It is recommended to create the command in the `initialize` function of the provider, and store them in a member named after the command, with the `_cmd` suffix. This makes it easy to reuse common settings throughout the provider.
 
-You can set default environment variables on the `@cmd.environment` Hash, and a default working directory using `@cmd.cwd=`.
+[TODO: it is usually best to avoid saying "It is recommended". Could you be more specific on whether they should do it or not?]
```

Comment:
Basically "if you do not have special needs, do this:" ; that's also why I put this in a note. There is little we can do to *force* provider writers to follow a certain pattern.

The reasoning for the recommendation is in the second sentence.